blockchain/types/src/crypto/metadata.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::BlockMetadataContents;
+use mc_crypto_digestible_signature::{DigestibleSigner, DigestibleVerifier};
+use mc_crypto_keys::{
+    Ed25519Pair, Ed25519Public, Ed25519Signature, Signature as SignatureTrait, SignatureError,
+};
+
+pub fn block_metadata_context() -> &'static [u8] {
+    b"block_metadata"
+}
+
+pub trait MetadataSigner: DigestibleSigner<Self::Signature, BlockMetadataContents> {
```

Comment:
I think I disagree from a Socratic perspective because the signing is a compact amount of code and likely won't deviate from that (or if it did become more complex, I'd think a separate area would then be warranted). However, I'm happy to go along with other opinions if there's a consensus on them.